client/pdo/client/controller/contract_controller.py

```diff
@@ -494,11 +596,30 @@ def do_exit(self, args) :
         """
         exit -- shutdown the simulator and exit the command loop
         """
+        if self.deferred > 0 : return False
+
+        try :
+            pargs = self.__arg_parse__(args)
+            parser = argparse.ArgumentParser(prog='exit')
+            parser.add_argument('-v', '--value', help='exit code', type=int, default=0)
+            options = parser.parse_args(pargs)
+
+            self.exit_code = options.value
+
+        except SystemExit as se :
```

Comment:
is there a missing return True/False statement within the try block? not sure i fully understand the exit logic

Reply:
sorry take this comment back